### PR TITLE
add: manual checkpoints

### PR DESCRIPTION
**Pull request by Robokishan**
Right now i am using this package with list of form. the problem is if we change input value, then also. history is getting recorded. i have a delete button. so I need a manual control over recording history. other packages have added manual checkpoint option. but those are required to use reducer. but in this package it is very easy so I have also added manual checkpoints in this also

if using manual checkpoint option then we need to pass `setInputList(list, true); ` true to record value in history.
other wise it is ignored. if we don't use manual checkpoint then it will work as expected

### CODE REVIEW

**Comment by homerchen19** [comment]
Hi @Robokishan I think this is a good feature.
Could you write some unit tests for it? `README` needs to be updated too.

File: index.ts [hunk 1]
```diff
@@ -33,65 +34,78 @@ const initialState = {
...
+          ? !!action.historyCheckpoint
+          : true;
+        const { newPresent } = action;
+
+        if (newPresent === present) {
+          return state;
+        }
+
+        return {
+          past: isNewCheckpoint == false ? past : [...past, present],
```
Comment (homerchen19) [inline index.ts:91]: Why don't we use `===`?

**Review by homerchen19 — CHANGES_REQUESTED** [review]

File: index.ts (hunk 1 above)
Comment (Robokishan) [reply index.ts:91]: Thanks will do that

**Comment by Robokishan** [comment]
> Hi @Robokishan I think this is a good feature. Could you write some unit tests for it? `README` needs to be updated too.

Sure will do that give me some time

**Comment by Robokishan** [comment]
@homerchen19 Can you please re review ?

**Comment by Robokishan** [comment]
@homerchen19 any update please ?

**Review by homerchen19 — APPROVED** [review]

**Pull request merged**